src/backend/gl/src/device.rs

```diff
@@ -788,6 +810,48 @@ impl d::Device<B> for Device {
             vertex_buffers[vb.binding as usize] = Some(*vb);
         }
 
+        let mut uniforms = Vec::new();
+        {
+            let gl = &self.share.context;
+            let mut count = 0;
+            let mut uniform_max_size = 0;
+
+            gl.GetProgramiv(program, gl::ACTIVE_UNIFORMS, &mut count);
+            gl.GetProgramiv(
+                program,
+                gl::ACTIVE_UNIFORM_MAX_LENGTH,
+                &mut uniform_max_size,
+            );
+            for uniform in 0..count {
+                let mut length = 0;
+                let mut size = 0;
+                let mut utype = 0;
+                let mut name = Vec::with_capacity(uniform_max_size as usize);
+                name.set_len(uniform_max_size as usize - 1);
+                gl.GetActiveUniform(
+                    program,
+                    uniform as _,
+                    uniform_max_size - 1 as i32,
+                    &mut length,
+                    &mut size,
+                    &mut utype,
+                    name.as_mut_ptr() as *mut _,
+                );
+
+                let location = gl.GetUniformLocation(program, name.as_ptr() as _);
+
+                // Sampler2D won't show up in UniformLocation and the only other uniforms
+                // should be push constants
+                if location >= 0 {
+                    uniforms.push(n::UniformDesc {
```

Comment:
Why is this code even a thing? Surely we can attain this info via SPIRV reflection?

Reply:
Due to my lack of knowledge of SPIRV reflection, I just looked at the spec and found that you cannot query Sampler2D uniforms the same as normal uniforms.

Reply:
Are you saying this can't be done with SPIRV reflection? If it can, than I'd rather it is.